src/EnergyPlus/ZoneContaminantPredictorCorrector.cc

```diff
@@ -1665,27 +1665,41 @@ namespace ZoneContaminantPredictorCorrector {
                 ControlledCO2ZoneFlag = false;
                 // Check all the controlled zones to see if it matches the zone simulated
                 for (ContControlledZoneNum = 1; ContControlledZoneNum <= NumContControlledZones; ++ContControlledZoneNum) {
-                    if (GetCurrentScheduleValue(ContaminantControlledZone(ContControlledZoneNum).AvaiSchedPtr) > 0.0) {
-                        ZoneAirCO2SetPoint = ZoneCO2SetPoint(ContaminantControlledZone(ContControlledZoneNum).ActualZoneNum);
-                        if (ContaminantControlledZone(ContControlledZoneNum).EMSOverrideCO2SetPointOn) {
-                            ZoneAirCO2SetPoint = ContaminantControlledZone(ContControlledZoneNum).EMSOverrideCO2SetPointValue;
+                    if (ContaminantControlledZone(ContControlledZoneNum).ActualZoneNum == ZoneNum) {
```

Comment:
So this loops over all contaminant controllers, looking for one that matches this zone.  This sort of backward referencing could require a lot more looping than what is actually needed.  It would be nice if there was some sort of forward referencing where we just loop over each existing controller and avoid all the empty steps.  But that's out of scope for here.